csharp/private/common.bzl

```diff
@@ -105,15 +47,17 @@ def fill_in_missing_frameworks(providers):
       deps: the original deps used to generate the providers.
     """
 
-    for gap_info in _find_framework_gaps(providers):
-        (base, compat_set_idx, start_idx, end_idx) = gap_info
+    # iterate through the compatability table since it's in preference order
+    for tfm in FrameworkCompatibility.keys():
```

Comment:
Haha this is even nicer than we talked about (doing 3 loops for each family.) Nice!